### PR TITLE
feat: pkg-utils-time

### PR DESCRIPTION
**Pull request by jxs1211**
Signed-off-by: xian-jie.shen <327411586@qq.com>

#### What type of PR is this?
ut 

#### What this PR does / why we need it:
ut for pkg

#### Special notes for your reviewer:
```sh
[going@dev utils]$ go test -cover time_test.go time.go
ok      command-line-arguments  0.003s  coverage: 100.0% of statements
[going@dev utils]$ cd ..
[going@dev pkg]$ cd ..
[going@dev crane]$ make lint
[going@dev crane]$ 
```

### CODE REVIEW

File: pkg/utils/time_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+			wantErr: false,
+		},
+		{
+			name:    "duration string's time unit is not 'd'",
+			s:       "300ms",
+			want:    300 * time.Millisecond,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
```
Comment (qmhu) [inline pkg/utils/time_test.go:29]: Would you please add a sample for a error case.

File: pkg/utils/time_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := ParseTimestamp(tt.ts)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ParseTimestamp() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if tt.name == "parsed incorrect" {
```
Comment (qmhu) [inline pkg/utils/time_test.go:71]: Please use another way to implement this.

`tt.name == "parsed incorrect"`

File: pkg/utils/time_test.go (hunk 1 above)
Comment (jxs1211) [reply pkg/utils/time_test.go:29]: done

File: pkg/utils/time_test.go (hunk 2 above)
Comment (jxs1211) [reply pkg/utils/time_test.go:71]: I've tried some other ways, but the `if != nil`'s situation will return a dynamic real date time, which makes the `want` value hard to be defined. Is there any other way to make a test case?

Comment (jxs1211) [reply pkg/utils/time_test.go]: done

File: pkg/utils/time_test.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	{
+		tt := tests[0]
+		t.Run(tt.name, func(t *testing.T) {
+			start := time.Now()
+			got, err := ParseTimestamp(tt.ts)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ParseTimestamp() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got.Nanosecond() <= start.Nanosecond() {
```
Comment (qmhu) [inline pkg/utils/time_test.go:79]: 可以判断下，如果有error就不用判断got和start了，因此可以把所有的case都用一个流程判断

**Review by qmhu — APPROVED** [review]

**Pull request merged**